.github/workflows/build.yml

```diff
@@ -0,0 +1,321 @@
+        
+  build-linux:
+    needs: [ get-go-version, get-product-version, generate-ld-flags ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [ linux ]
+        goarch: [ "arm", "arm64", "386", "amd64" ]
+      fail-fast: true
+    
+    name: Go ${{ needs.get-go-version.outputs.go-version }} ${{ matrix.goos }} ${{ matrix.goarch }} build
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+     
+      - name: Build dependencies
+        run: make deps
+      
+      - name: Setup node and yarn
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+          cache-dependency-path: 'ui/yarn.lock'
+      
+      - name: Install Yarn
+        run: |
+          npm install -g yarn
+      
+      - name: Build prerelease
+        run: make prerelease
+      
+      - name: Install Linux build utilties
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y software-properties-common
+          sudo dpkg --add-architecture i386
+          sudo apt-get update
+          sudo apt-get install -y \
+            libc6-dev-i386 \
+            libpcre3-dev \
+            linux-libc-dev:i386 
+          sudo apt-get install -y \
+            binutils-aarch64-linux-gnu \
+            binutils-arm-linux-gnueabihf \
+            gcc-aarch64-linux-gnu \
+            gcc-arm-linux-gnueabihf \
+            gcc-multilib-arm-linux-gnueabihf
+          sudo apt-get upgrade -y
+
+      - name: Set gcc
+        run: |
+          if [ "${{ matrix.goarch }}" == "arm" ]; then
+            echo "CC=arm-linux-gnueabihf-gcc" >> $GITHUB_ENV
+          elif [ "${{ matrix.goarch }}" == "arm64" ]; then
+            echo "CC=aarch64-linux-gnu-gcc" >> $GITHUB_ENV
+          fi
+
+      - name: Build
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+          GO_TAGS: "${{ env.GO_TAGS }}"
+          GO_LDFLAGS: ${{ needs.generate-ld-flags.outputs.ldflags }}
+          CGO_ENABLED: 1
+        run: |
+          make pkg/${{ matrix.goos }}_${{ matrix.goarch }}.zip
+          mv pkg/${{ matrix.goos }}_${{ matrix.goarch }}.zip ${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+          path: pkg/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+    
+      - name: Package
+        if: ${{ matrix.goos == 'linux' }}
+        uses: hashicorp/actions-packaging-linux@v1
+        with:
+          name: ${{ env.PKG_NAME }}
```

Comment:
This is looking really good!! On line 204, `name` should be set to `nomad-enterprise` instead of env.PKG_NAME which just evaluates to `nomad`. In vault-ent, we set the value like this: https://github.com/hashicorp/vault-enterprise/blob/main/.github/workflows/build.yml#L180. You can do a similar thing here, or just tack `-enterprise` onto the end there. If we plan to add nightlies in the future, I'd recommend tacking on `-enterprise` (since `${{ github.event.repository.name }}` is not set when the build workflow is called on a schedule)

Reply:
Great catch @mdeggies! Will update this

Reply:
Does this needs to be set to `nomad-enterprise`here as well? I made a similar comment in the [ENT repo](https://github.com/hashicorp/nomad-enterprise/pull/708#discussion_r829569074)

Reply:
I've updated this PR to use `name: ${{ env.PKG_NAME }}-enterprise`

Reply:
Yep, Claire this looks 100% right to me. For OSS it should be `${{ env.PKG_NAME }}`, and for ent we just need to tack on `-enterprise` at the end.

Reply:
> For OSS it should be ${{ env.PKG_NAME }}, and for ent we just need to tack on -enterprise at the end.

So this PR is not right, it's using [`name: ${{ env.PKG_NAME }}-enterprise`](https://github.com/hashicorp/nomad/pull/12276/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R204) while the ENT repo is using only [` name: ${{ env.PKG_NAME }}`](https://github.com/hashicorp/nomad-enterprise/pull/708/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R226)

Reply:
Yeah that's my bad -- I got confused, but will remove it to use just `name: ${{ env.PKG_NAME }}` on this OSS repo and update the ENT to use `name: ${{ env.PKG_NAME }}-enterprise`